### ISSUE TITLE
 extraneous parentheses

### ISSUE THREAD

**Issue by RetoFx** [STMicroelectronics/stm32f1xx_hal_driver#13]
Hi
Mi ARMCLang report in file stm32f1xx_hal_usart.c Line 2785

if((husart->Instance == USART1))

../../HAL/STM32/STM32F1xx/stm32f1xx_hal_usart.c(2785): warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
  if((husart->Instance == USART1))
      ~~~~~~~~~~~~~~~~~^~~~~~~~~
../../HAL/STM32/STM32F1xx/stm32f1xx_hal_usart.c(2785): note: remove extraneous parentheses around the comparison to silence this warning
  if((husart->Instance == USART1))
     ~                 ^        ~
../../HAL/STM32/STM32F1xx/stm32f1xx_hal_usart.c(2785): note: use '=' to turn this equality comparison into an assignment

Please replace it with 

if(husart->Instance == USART1)

**Comment by ASEHSTM** [comment]
Hello @RetoFx,

Thank you for your contribution. This will be fixed soon.

With regards,

**Comment by ASEHSTM** [comment]
ST Internal Reference: 182974

**Comment by TOUNSTM** [comment]
fixed in 6ca7fc40cc862623dc05b33b0014789c1fbbbb86
